File: itmd-521/labs/chapter-05/src/main/python/lab900_simple_lambda/simpleLambdaApp.py
"""
 Simple Python lambda  usage application.

 @author rambabu.posa
"""
# a few French first names that can be composed with Jean
frenchFirstNameList = ["Georges", "Claude","Philippe", "Pierre", "Francois", "Michel", "Bernard",
                       "Guillaume", "Andre", "Christophe", "Luc", "Louis"]

for name in frenchFirstNameList:
    print("{}  and Jean-{}  are different French first names!".format(name,name))

print("---------------------------------")

def foo(name):
    print("{}  and Jean-{}  are different French first names!".format(name,name))

for name in frenchFirstNameList:
    foo(name)


print("---------------------------------")
f = lambda name: name+" and Jean-"+name+"  are different French first names!"

for name in frenchFirstNameList:
    print(f(name))

File: itmd-521/labs/chapter-05/src/main/java/net/jgp/books/spark/ch05/lab900_simple_lambda/SimpleLambdaApp.java
package net.jgp.books.spark.ch05.lab900_simple_lambda;

import java.util.Arrays;
import java.util.List;

public class SimpleLambdaApp {

  public static void main(String[] args) {
    // a few French first names that can be composed with Jean
    List<String> frenchFirstNameList = Arrays.asList("Georges", "Claude",
        "Philippe", "Pierre", "François", "Michel", "Bernard", "Guillaume",
        "André", "Christophe", "Luc", "Louis");

    frenchFirstNameList.forEach(
        name -> System.out.println(name + " and Jean-" + name
            + " are different French first names!"));

    System.out.println("-----");

    frenchFirstNameList.forEach(
        name -> {
          String message = name + " and Jean-";
          message += name;
          message += " are different French first names!";
          System.out.println(message);
        });
  }
}
